### PR TITLE
RecognizedByParents - adding back pointer to the connection in hub proxy

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: src/signalrclient/hub_connection_impl.cpp [hunk 1]
```diff
@@ -70,7 +70,8 @@ namespace signalr
             return iter->second;
         }
 
-        auto proxy = std::make_shared<internal_hub_proxy>(hub_name, m_logger);
+        auto weak_connection = std::weak_ptr<hub_connection_impl>(shared_from_this());
```
Comment (jordanrudd) [inline src/signalrclient/hub_connection_impl.cpp]: Is the weak_ptr here to prevent this method from blocking the connection from getting destroyed?

**Comment by jordanrudd** [comment]
Looks good!

File: src/signalrclient/hub_connection_impl.cpp (hunk 1 above)
Comment (moozzyk) [inline src/signalrclient/hub_connection_impl.cpp]: Yes, it's to prevent cycles.

**Comment by moozzyk** [comment]
Thanks @jordanrudd !

**Pull request merged**